examples/quantum_associative_memory.cpp

```diff
@@ -48,7 +48,7 @@ int main()
     bool bit;
     std::cout << "Learning (Two's complement)..." << std::endl;
     for (perm = 0; perm < InputPower; perm++) {
-        std::cout << "Epoch " << (perm + 1U) << " out of " << InputPower << std::endl;
+        std::cout << "Epoch " << (uint64_t)(perm + 1U) << " out of " << (uint64_t)InputPower << std::endl;
```

Comment:
This seems problematic - does C++ operator<< not support 128bit variables?  And if that's really the case, consider adding an operator<<(uint128_t) overload so that these don't have to be explicitly cast everywhere.

Reply:
Done.